service/history/historyEngine.go

```diff
@@ -518,12 +538,42 @@ func (e *historyEngineImpl) StartWorkflowExecution(
 		RunId:      common.StringPtr(uuid.New()),
 	}
 	clusterMetadata := e.shard.GetService().GetClusterMetadata()
-	mutableState, err := e.createMutableState(clusterMetadata, domainEntry, workflowExecution.GetRunId())
+	curMutableState, err := e.createMutableState(clusterMetadata, domainEntry, workflowExecution.GetRunId())
 	if err != nil {
 		return nil, err
 	}
 
-	startEvent, err := mutableState.AddWorkflowExecutionStartedEvent(
+	// for signalWithStart
+	isSignalWithStart := signalWithStartArg != nil
+	var prevMutableState execution.MutableState
+	if isSignalWithStart && signalWithStartArg.prevMutableState != nil {
+		prevMutableState = signalWithStartArg.prevMutableState
+	}
+	if prevMutableState != nil {
```

Comment:
Could we add some comment about the behavior of signalWithStart? The case I am not clear is when the workflow is closed and no current open workflow. When calling signalWithStart, what is going to happen?

Reply:
It will start workflow according to WorkflowIDReusePolicy, default policy for signalwithstart is WorkflowIDReusePolicyAllowDuplicate (which is different from startWorkflow)

Reply:
updated with comment